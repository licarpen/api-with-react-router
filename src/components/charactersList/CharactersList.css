.CharactersList{
  margin: auto;
  text-align: center;
  font-family: Arial, Helvetica, sans-serif;

  ul{
    display: flex;
    flex-wrap: wrap;
  }

  li{
    list-style: none;
    padding: 15px;
  }

}