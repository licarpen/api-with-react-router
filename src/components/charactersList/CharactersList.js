import React from 'react';
import styles from './CharactersList.css';
import { useCharacters } from '../../hooks/charactersList';

const CharactersList = () => {
  const { charactersArray } = useCharacters();

  return (
    <div className={styles.CharactersArray}>
      { charactersArray.length ? 
        charactersArray.map(({ name, image }) => { 
          return <div key={name}>
            <h2>{name}</h2>
            <img src={image}/>
          </div>;}) : null
      }
    </div>
  );
};

export default CharactersList;
