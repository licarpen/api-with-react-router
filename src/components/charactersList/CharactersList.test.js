import React from 'react';
import { shallow } from 'enzyme';
import CharactersList from './CharactersList';

describe('CharactersList component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<CharactersList />);
    expect(wrapper).toMatchSnapshot();
  });
});
