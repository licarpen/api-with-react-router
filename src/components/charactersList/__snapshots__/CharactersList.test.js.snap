// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharactersList component matches a snapshot 1`] = `
<div
  className="CharactersArray"
/>
`;
